src/bytes.rs

```diff
@@ -2,12 +2,12 @@ use crate::{Buf, BufMut, IntoBuf};
 use crate::buf::IntoIter;
 use crate::debug;
 
-use std::{cmp, fmt, mem, hash, slice, ptr, usize};
-use std::borrow::{Borrow, BorrowMut};
-use std::ops::{Deref, DerefMut, RangeBounds};
-use std::sync::atomic::{self, AtomicUsize, AtomicPtr};
-use std::sync::atomic::Ordering::{Relaxed, Acquire, Release, AcqRel};
-use std::iter::{FromIterator, Iterator};
+use core::{cmp, fmt, mem, hash, slice, ptr, usize};
+use alloc::borrow::{Borrow, BorrowMut};
```

Comment:
alloc is imported only on no_std target, do you know if it is always implicitly imported on std?
AFAIK in past I remember I needed to `use std as alloc` to make it work

Reply:
thanks, I guess the situation is more complex than I realized